soc/xtensa/intel_adsp/ace/include/intel_ace15_mtpm/adsp_power.h

```diff
@@ -43,4 +46,20 @@ struct ace_pwrsts {
 
 #define ACE_PWRSTS ((volatile struct ace_pwrsts *) &ACE_DfPMCCU.dfpwrsts)
 
+static ALWAYS_INLINE void soc_cpu_power_up(int cpu_num)
+{
+	ACE_PWRCTL->wpdsphpxpg |= BIT(cpu_num);
+}
+
+static ALWAYS_INLINE void soc_cpu_power_down(int cpu_num)
+{
+	ACE_PWRCTL->wpdsphpxpg &= ~BIT(cpu_num);
+}
+
+static ALWAYS_INLINE bool soc_cpu_is_powered(int cpu_num)
```

Comment:
Aren't these names a bit too generic? I mean no `ace_` prefix/namespace or anything.

Reply:
I could change these if you insist. I made these generic so we don't need to rename stuff if we ever need to share code with another audio DSP architecture in the future.

Reply:
> I could change these if you insist. I made these generic so we don't need to rename stuff if we ever need to share code with another audio DSP architecture in the future.

Thanks, I was wondering if this was by design. If this is generic by design then great but then leave a short comment that it's meant to be. There is zero comment right now.

Reply:
Added the function description (doxygen). Hopefully this would be enough.

Unless we have a porting guide, it's hard to convey the idea of "generic function name to ease future platform addition" in the comment without it sounding out of place.

Reply:
Thanks, the doxygen looks useful but that's not what I had in mind.

> it's hard to convey the idea of "generic function name to ease future platform addition" in the comment 

I think you just did, brilliantly :-)

> without it sounding out of place.

I don't see why it would sound out of place.

It's not important, nevermind.

Reply:
I opt for including this in the commit message.